### PR TITLE
laercionit.md

### PR DESCRIPTION
**Pull request by laercionit**
[02/01/2015]

I hereby agree to the terms of the Contributors License
Agreement, version 1.0, with MD5 checksum
2a87b790465ff9669faa4253a922846068a92ac4

I furthermore declare that I am authorized and able to make this
agreement and sign this declaration.

Signed,

[Laercio Mendonça]
https://github.com/laercionit

### CODE REVIEW

**Comment by dasgarner** [comment]
Hi Laercio,

I've just had a quick look and in your xibo-cms fork it says you have changed **31900 lines of code in 147 files** - I don't think this can be correct? I can also see files in the commit that really shouldn't be there - like the entire bootstrap source code?

When we make changes like this we need to make sure that they have the smallest possible impact to the code - otherwise we end up with a support nightmare and many unhappy users...


The CLA only needs to be in xibosignage/xibo, which I see you have done in a separate pull request - that is great.

File: theme/default/html/grid_render.php [hunk 1]
```diff
@@ -35,6 +35,7 @@
     <div class="widget-body">
         <?php echo Theme::Get('prepend'); ?>
         <div class="XiboGrid" id="<?php echo Theme::Get('id'); ?>">
+        	<?php echo Theme::Get('append'); ?>
```
Comment (dasgarner) [inline theme/default/html/grid_render.php]: I dont understand why this has been moved? Append should be after the data grid?

File: theme/default/js/neon-custom.js [hunk 2]
```diff
@@ -0,0 +1,2082 @@
+/**
+ *	Neon Main JavaScript File
+ *
+ *	Theme by: www.laborator.co
+ **/
```
Comment (dasgarner) [inline theme/default/js/neon-custom.js]: What is the licence on this file?

**Comment by laercionit** [comment]
Dan , i copy directory from my developer server to my git directory
checkout.


Em segunda-feira, 2 de fevereiro de 2015, Dan Garner <
notifications@github.com> escreveu:

> Hi Laercio,
>
> I've just had a quick look and in your xibo-cms fork it says you have
> changed *31900 lines of code in 147 files* - I don't think this can be
> correct? I can also see files in the commit that really shouldn't be there
> - like the entire bootstrap source code?
>
> When we make changes like this we need to make sure that they have the
> smallest possible impact to the code - otherwise we end up with a support
> nightmare and many unhappy users...
>
> The CLA only needs to be in xibosignage/xibo, which I see you have done in
> a separate pull request - that is great.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xibosignage/xibo-cms/pull/17#issuecomment-72553460>.
>


-- 
*Laercio Junior*
Cel./Pessoal: (21) 99162-7440
Email Pessoal: laercionit@gmail.com
MSN: laercionit@hotmail.com
Skype: laercionit

File: theme/default/js/scripts.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+$(document).ready(function() {
```
Comment (dasgarner) [inline theme/default/js/scripts.js]: What does this do?

Should this be in English and served through Xibo's translations?

File: theme/default/libraries/flot/LICENSE.txt [hunk 4]
```diff
@@ -0,0 +1,22 @@
+Copyright (c) 2007-2013 IOLA and Ole Laursen
```
Comment (dasgarner) [inline theme/default/libraries/flot/LICENSE.txt]: We only just put in morrisjs as we deemed it to be better than flot

File: theme/default/js/neon-custom.js (hunk 2 above)
Comment (laercionit) [inline theme/default/js/neon-custom.js]: I think that my web designer using our caralog templates designer .


Em segunda-feira, 2 de fevereiro de 2015, Dan Garner <
notifications@github.com> escreveu:

> In theme/default/js/neon-custom.js
> <https://github.com/xibosignage/xibo-cms/pull/17#discussion_r23966250>:
>
> > @@ -0,0 +1,2082 @@
> > +/**
> > + *	Neon Main JavaScript File
> > + *
> > + *	Theme by: www.laborator.co
> > + **/
>
> What is the licence on this file?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/xibosignage/xibo-cms/pull/17/files#r23966250>.
>


-- 
*Laercio Junior*
Cel./Pessoal: (21) 99162-7440
Email Pessoal: laercionit@gmail.com
MSN: laercionit@hotmail.com
Skype: laercionit

**Comment by dasgarner** [comment]
Sorry, but I cannot merge that as it stands - there are simply too many changes for me to accurately work out what is what.

Lots of questions too.

I will have a think about how we can proceed with this - I thought the goal was to make the sidebar responsive :smile: not rewrite the entire front end


I'm sorry if I sound like im being difficult, but changing this much in one go will make it impossible to tell if anything broke!

**Comment by laercionit** [comment]
Dan, do not need to apologize.
I'll call Bruno and explain the mistakes he made. You know designer, sees
something that can improve and changes everything.

I'll have to just fit the side menu for you.

On the subject, had the opportunity to put into execution to analyze?

*Laercio Junior*
Cel./Pessoal: (21) 99162-7440
Email Pessoal: laercionit@gmail.com
MSN: laercionit@hotmail.com
Skype: laercionit



2015-02-02 20:51 GMT-02:00 Dan Garner <notifications@github.com>:

**Comment by laercionit** [comment]
Dan, I understood the problem.

1-My team used a responsive theme of our catalog (purchased) to customize
the default theme, some scripts. I have given the scolding he needed to
.... lol

2- The GIT does not analyze the code as well as Subversion, or copy a file
without modifying it interprets as change.

3- On the question of how to remove some columns of the GRID to the mobile
version, some columns are unnecessary, generating horizontal scrolling.

Example: In Layout (ID, Owner, permissions and status), but would remove
only theme for my internal use.


4- I will speak to Bruno to install GIT and do locally. By default we have
some environments for development, approval and production.

He did everything locally, put in development and after approval in
approval. I copied the directory of the approval server and put in my
checkout GIT. The GIT understood that everything had copied modification
without analyzing the file contents. Here we use subversion

**Comment by dasgarner** [comment]
Hi @laercionit 

I did try out the theme but unfortunately it doesn't render correctly at the moment :smile: 

I did notice that you have submitted the CLA to this repository - but it will need to go in the xibosignage/xibo one as that is the central location that covers all projects. I thought I saw a pull request from you last night, but it has disappeared.

To move this forward I think we need to do the following things:
1. Revert to a base xibosignage/xibo develop branch
2. Create and checkout a branch called "feature/default-theme-improvement"
2. Revert flot.js charting
3. Revert inclusion of non minified library source files
4. Revert the use of googlefonts api (wont work offline)
5. Remove the CLA and put in xibosignage/xibo
6. Remove the proprietary theme CSS file, unless its licence can be changed to MIT or similar

Once we do that I think we will have a much smaller change log to deal with, which we can see exactly what modifications have been made and comment accordingly.

I hope that sounds OK?

**Pull request closed without merging**